invoice/src/main/java/org/killbill/billing/invoice/generator/InvoiceDateUtils.java

```diff
@@ -26,7 +26,9 @@
 import org.joda.time.Weeks;
 import org.joda.time.Years;
 import org.killbill.billing.catalog.api.BillingPeriod;
+import org.killbill.billing.util.config.definition.InvoiceConfig;
 import org.killbill.billing.util.currency.KillBillMoney;
+import org.killbill.commons.utils.annotation.VisibleForTesting;
 
 public class InvoiceDateUtils {
```

Comment:
This class currently only offers `static` method, because there is no state that is kept. However, now that we need the config (or at least one param of the config), this makes less sense. It would be better to have a real instance - perhaps injected with guice, that takes a CTOR with the `InvoiceConfig`, in such a way that we don't mix request parameters (`startDate`, ...) and static state from the system (`config`).

1. Could you experiment to see if this is not too much trouble doing this - I think the result would be much cleaner
2. If this proves difficult, then at least let's not pass the `InvoiceConfig`, but instead just the Integer for the `prorationFixedDays`

Reply:
As discussed, I've updated the methods in `InvoiceDateUtils` to accept `prorationFixedDays` instead of `invoiceConfig`.